cmd/device-worker/main.go

```diff
@@ -97,14 +99,21 @@ func main() {
 	wl.RegisterObserver(dataMonitor)
 	dataMonitor.Start()
 
-	hbs := heartbeat2.NewHeartbeatService(c, configManager, wl, &hw, dataMonitor)
+	hbs := heartbeat2.NewHeartbeatService(dispatcherClient, configManager, wl, &hw, dataMonitor)
+
 	configManager.RegisterObserver(hbs)
 
+	dr, err := deregistration2.NewDeregistration(wl, configManager, hbs)
+	if err != nil {
+		log.Fatal(err)
+	}
+	configManager.RegisterObserver(dr)
```

Comment:
`Deregistration` is registered as a configuration observer and Update method there is responsible for deregistering the device, which is not a desired behavior - the device will get de-registered as soon as it receives the first configuration update.